From dc37a8420816ad43b9c91d75aafcfc99d2cd78f8 Mon Sep 17 00:00:00 2001
From: Stuart Menefy <stuart.menefy@codasip.com>
Date: Mon, 20 May 2024 16:15:52 +0100
Subject: [PATCH 17/17] target/riscv: Fix pmu_mask and pmp default for Codasip
 A730

Values set in the ..._cpu_init() functions for the CPU configuration,
which can also be set via properties, will be overridden when the
CPU is realised and properties assigned. If we want to change
the default values, it is necessary to change the default values
in the properties.
---
 target/riscv/cpu.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/target/riscv/cpu.c b/target/riscv/cpu.c
index 4a312d3bde..24e8f3f8f9 100644
--- a/target/riscv/cpu.c
+++ b/target/riscv/cpu.c
@@ -590,6 +590,17 @@ static void rv64_codasip_a730_cpu_init(Object *obj)
     /* Ssu64xl */
     /* Sdext */
     /* Sdtrig */
+
+    /*
+     * The cfg.pmu_mask and cfg.pmp set above will be overridden when the
+     * propertities are added to CPU object and the default values set.
+     */
+    for (Property *prop = riscv_cpu_options; prop && prop->name; prop++) {
+        if (!strcmp(prop->name, "pmu-mask"))
+            prop->defval.i = MAKE_64BIT_MASK(3, 4);
+        if (!strcmp(prop->name, "pmp"))
+            prop->defval.i = false;
+    }
 }
 
 static void rv128_base_cpu_init(Object *obj)
-- 
2.39.3

